internal/host/hostutil/host_utils.go

```diff
@@ -193,3 +195,22 @@ func UnmarshalConnectivityReport(reportStr string) (*models.ConnectivityReport,
 	}
 	return &report, nil
 }
+
+func GetHostCluster(log logrus.FieldLogger, db *gorm.DB, host *models.Host) (*common.Cluster, error) {
+	var cluster common.Cluster
+	err := db.First(&cluster, "id = ?", host.ClusterID).Error
+	if err != nil {
+		log.WithError(err).Errorf("Failed to find cluster %s", host.ClusterID)
+		return nil, errors.Errorf("Failed to find cluster %s", host.ClusterID)
+	}
+	return &cluster, nil
+}
+
+func IsSingleNode(log logrus.FieldLogger, db *gorm.DB, host *models.Host) bool {
+	cluster, err := GetHostCluster(log, db, host)
+	if err != nil {
+		log.Debug("Can't check if host is part of single node OpenShift")
```

Comment:
log.WithError?

Reply:
The error is already logged in GetHostCluster